packages/injective-math/src/fp_decimal/trigonometry.rs

```diff
@@ -0,0 +1,132 @@
+use num::pow::Pow;
+
+use crate::fp_decimal::FPDecimal;
+
+impl FPDecimal {
+    pub fn _cos(mut x: FPDecimal) -> FPDecimal {
+        x = FPDecimal::_change_range(x);
+        FPDecimal::_sin(FPDecimal::PI / FPDecimal::TWO - x)
+        /*
+        let sign = x.sign;
+        x.sign = 1;
+        let mut output = FPDecimal::ONE - (x.pow(FPDecimal::TWO) / FPDecimal::TWO.factorial())
+            + (x.pow(FPDecimal::FOUR) / FPDecimal::FOUR.factorial())
+            - (x.pow(FPDecimal::SIX) / FPDecimal::SIX.factorial())
+            + (x.pow(FPDecimal::EIGHT) / FPDecimal::EIGHT.factorial())
+            - (x.pow(FPDecimal::TWO + FPDecimal::EIGHT) / (FPDecimal::TWO + FPDecimal::EIGHT).factorial())
+            + (x.pow(FPDecimal::FOUR + FPDecimal::EIGHT) / (FPDecimal::FOUR + FPDecimal::EIGHT).factorial());
+        output.sign = sign;
+        output
+        */
+    }
+
+    fn _taylro_expansion(x: FPDecimal) -> FPDecimal {
```

Comment:
wrong name? ->`_taylor_expansion`, also why stop at exactly 9 ?

Reply:
1. 9 provides a good enough approximation to compute the sine(x), the relative error is less than 1%,
2. using a number greater than 9 may have an overflow issue.